azure-iot-device/azure/iot/device/common/errors_paho.py

```diff
@@ -0,0 +1,165 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+class OperationCancelledError(Exception):
+    """
+    Operation was cancelled.
+    """
+
+    pass
+
+
+class PahoConnectionRefusedProtocolVersionError(Exception):
+    """
+    Connection Refused: unacceptable protocol version.
+    """
+
+    pass
+
+
+class PahoConnectionRefusedIdentifierRejectedError(Exception):
```

Comment:
>PahoConnectionRefusedIdentifierRejectedError [](start = 6, length = 44)

Do you think it will be a good idea to suggest resolution for these type of error on the docstrings ? Like for this specific one it is not very clear from the name of the `CONNACK_REFUSED_IDENTIFIER_REJECTED` as well. We can suggest something like this can happen due to many reasons like 
1. `client_id` not present for paho. Make sure device_id and/or module_id is present
2. `client_id` wrong for paho

Reply:
I like that idea, but I think that's better added after we get an idea of what can cause the errors we have.  Right now we're still narrowing down the specific errors.

---
In reply to: [302699577](https://github.com/Azure/azure-iot-sdk-python-preview/pull/136#discussion_r302699577) [](ancestors = 302699577)